src/main/scala/com/advancedtelematic/campaigner/db/Campaigns.scala

```diff
@@ -153,20 +153,24 @@ protected [db] class Campaigns(implicit db: Database, ec: ExecutionContext)
    * these campaigns, selects the most recent failures and returns them.
    */
   protected[db] def findFailedDeviceUpdatesAction(campaignIds: Set[CampaignId]): DBIO[Set[DeviceUpdate]] = {
-    Schema.deviceUpdates
-      .join(Schema.deviceUpdates
-      .filter(upd => upd.campaignId.inSet(campaignIds))
+    val latestFinishedUpdates =
+      Schema.deviceUpdates
+        .filterNot(_.status inSet Set(DeviceStatus.requested, DeviceStatus.scheduled, DeviceStatus.accepted))
```

Comment:
What was the reason to change it from `.filter(_.status inSet Set(DeviceStatus.successful, ...)`? I personally think it looked better the previous way.

Reply:
_Side note_: we can also have `val finishedStates = Set(DeviceState.successful ...)` in `DeviceState` object as these states are used quite a lot throughout the codebase.

Reply:
It was simply for convenience :) There were less to exclude than to include. So the equivalent would be:
```.filter(_.status inSet Set(DeviceStatus.rejected, DeviceStatus.successful, DeviceStatus.cancelled, DeviceStatus.failed))```
You might be right about that being more readable. I could change it back.

Reply:
About the side note: yes, that might be useful indeed. It would also give us a centralised definition of "finished". Right now it's scattered and maybe not consistent. We can refactor that in a follow-up PR.

Reply:
I've updated to use `filter` instead of `filterNot` to improve readability.